lib/models/archimedes/functions/question.rb

```diff
@@ -3,11 +3,46 @@ class Question < Archimedes::Function
     def call
       query, *_ = @args
 
-      response = Magma::Client.instance.query(
-        @token,
-        @project_name,
-        query.to_values
-      )
+      host = Timur.instance.config(:magma).fetch(:host)
+
+      client = Etna::Client.new(
+        host,
+        @token)
+      
+      query_route = client.routes.find { |r| r[:name] == 'query' }
+
```

Comment:
You'll have to verify this, but the client should already come with route methods built-in so you can just call `client.query(*query_params)`. I also don't think any of the hmac-signing is required since these routes can just be authenticated using a token.